File: week1/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length < 1)
            return "";
        // Set prefix to first string in strs
        StringBuilder prefix = new StringBuilder(strs[0]);
        for (String s : strs) {
            if (prefix.length() < 1)
                return "";
            // Truncate prefix if necessary
            if (prefix.length() > s.length()) {
                prefix.delete(s.length(), prefix.length());
            }
            // Iterate through string s until prefix differs
            for (int i = 0; i < s.length(); i++) {
                if (prefix.length() <= i || s.charAt(i) != prefix.charAt(i)) {
                    prefix.delete(i, prefix.length());
                    break;
                }
            }
        }
        return prefix.toString();
    }
}

File: spring2020/unit1/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ""
        if len(strs) < 1:
            return res
        for i in range(len(strs[0])):
            next_letter = None
            for j in range(len(strs)):
                if len(strs[j]) <= i:
                    return res
                if not next_letter:
                    next_letter = strs[j][i]
                if strs[j][i] != next_letter:
                    return res
            res += next_letter
        return res
